src/buyflow/Buyflow.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react'
+import AgeStep from './AgeStep'
+import EmailStep from './EmailStep'
+import SummaryStep from './SummaryStep'
+
+interface BuyflowProps {
+  productId: ProductIds
+}
+
+export enum ProductIds {
+  devIns = 'dev_ins',
+}
+
+const PRODUCT_IDS_TO_NAMES = {
+  [ProductIds.devIns]: 'Developer Insurance',
+}
+
+const Buyflow: React.FC<BuyflowProps> = (props) => {
+  const [currentStep, setStep] = useState('email')
+  const [collectedData, updateData] = useState({
+    email: '',
+    age: 0,
+  })
+  const getStepCallback = (nextStep: string) => (field: string, value: any) => {
+    updateData({ ...collectedData, [field]: value })
+    setStep(nextStep)
+  }
+  return (
+    <>
+      <h4>Buying {PRODUCT_IDS_TO_NAMES[props.productId]}</h4>
+      {(currentStep === 'email' && <EmailStep cb={getStepCallback('age')} />) ||
+        (currentStep === 'age' && (
+          <AgeStep cb={getStepCallback('summary')} />
+        )) ||
+        (currentStep === 'summary' && (
+          <SummaryStep collectedData={collectedData} />
+        ))}
```

Comment:
Hey. Starting with the most important thing: since the idea of this component is to be able to adapt flow for different scenarios, we need to unhardcode the sequence of steps and move them to the product configuration.

In the buying flow component, we would receive the product id as a prop (router will handle it), get the product configuration by this id, and render UI that would correctly react on user input.

This means that the selected code would become looking like:
```ts
<StepComponent cb={getStepCallback()} collectedData={collectedData} />
```
note that we are not breaking step components API which means current code would just continue to work